### DISCUSSION TITLE
DietPi-Software | RustDesk server

### DISCUSSION THREAD

**Discussion by Niaz89** [MichaIng/DietPi#5656] [Software Request]
#### Formal software information
- RustDesk Server
- FOSS virtual/remote desktop server for RustDesk clients
- https://rustdesk.com/server/
- https://github.com/rustdesk/rustdesk-server
- https://rustdesk.com/docs/en/self-host/install/

#### Are there similar/alternative software titles available with DietPi-Software?

- NoMachine

#### What makes your requested software better than the above solutions, if available?

- It is fully open source, without any (feature) limitation to free users. AFAIK it doesn't even provide any priced plans.

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?

- ...

#### Can you provide the installation steps that you would suggest DietPi-Software to do?

- Unfortunately, no.

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.

- Unfortunately, I can't be of any help.

**Comment by Joulinar** [comment]
I moved the request to GitHub discussions. Vote for this software implementation with the ⬆️ button on this post.

**Comment by marcdraco1-gif** [comment]
Well that's weird. I'm logged in. RustDesk is chasing the mighty $$$ now but I think we should be OK with Dietpi on an SBC so long as the server does the initial handshake and then the peers connect to each other.

**Reply by MichaIng** [reply]
> RustDesk is chasing the mighty $$$

Did something change recently their end?

**Reply by marcdraco1-gif** [reply]
It's still Open Source but it looks like you have to run your own server to make it operate. Perhaps I got that wrong - it's apparently a "clone" of Distant Desktop from what I've seen. I moved to that after another famous one (damn old age and my memory) started cutting us off after a minute or two. DD allows me hours per month which is more than enough for me to attend to my (even more elderly friends!) when they get themselves into a pickle.

If we need a server at one end, I'll happily donate them some old RPis to run DP in their homes so I can drop in when needs be.

**Reply by MichaIng** [reply]
As far as I can see, it still works for free without an own server. However, bandwidth is limited. There is a notice since a long time at the bottom that a self-hosted server is needed for faster connection. This request is about implementing the installation of such connection server.

The free OSS variant of this server seems to be without dedicated admin web interface, while the pro version which costs money has a web interface for easier administration.

So yeah, this OP statement is not true anymore

> without any (feature) limitation to free users. AFAIK it doesn't even provide any priced plans.

@Niaz89 do you remember whether the OSS server ever had any of the features the pro version has now? Or are the pro features entirely new, so that the OSS server is still as usable as it was when you opened the request?

**Reply by JappeHallunken** [reply]
The OSS versions lacks of this features, compared to the PRO version:

- Account
- [Web console](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/)
- [API](https://github.com/rustdesk/rustdesk/wiki/FAQ#api-of-rustdesk-server-pro)
- [OIDC](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/oidc/), [LDAP](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/ldap/), [2FA](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/2fa/)
- Address book
- Log management (Connection, file transfer, alarm, etc.)
- Device management
- [Security Settings sync](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/strategy/)
- [Access control](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/permissions/)
- [Multiple relay servers](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/relay/) (automatically selects your closest relay)
- [Custom client generator](https://rustdesk.com/docs/en/self-host/client-configuration/#1-custom-client-generator-pro-only)
- WebSocket
- Web client self-host

[https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/)

But there are also different plans with different feature sets:
[https://rustdesk.com/pricing/](https://rustdesk.com/pricing/)

**Comment by MichaIng** [comment]
PR up: #7842

**Comment by StephanStS** [comment]
Implemented with release v9.20: https://dietpi.com/docs/releases/v9_20/
